added recurring execution entity

pondrejk opened the pull request:
Related to #400 

```
In [5]: rec=entities.RecurringLogic(id=2)

In [6]: rec.read_json()
Out[6]: 
{u'cron_line': u'* * * * *',
 u'end_time': None,
 u'id': 2,
 u'iteration': 2,
 u'state': u'finished',
 u'task_group_id': 8,
 u'tasks': [{u'cli_example': None,
   u'ended_at': u'2017-04-11 11:30:12 UTC',
   u'humanized': {u'action': u'Run hosts job:',
    u'errors': [u'A sub task failed'],
    u'input': u'Run ls',
    u'output': u'2 task(s), 0 success, 2 fail'},
   u'id': u'cfbc701b-0909-4b3f-aa3b-0878df8437dd',
   u'input': {u'current_user_id': 3,
    u'job_category': u'Commands',
    u'job_invocation': {u'description': u'Run ls',
     u'id': 6,
     u'name': u'Commands'},
    u'job_invocation_id': 6,
    u'locale': u'en'},
   u'label': u'Actions::RemoteExecution::RunHostsJob',
   u'output': {u'failed_count': 2,
    u'pending_count': 0,
    u'success_count': 0,
    u'total_count': 2},
   u'pending': False,
   u'progress': 1.0,
   u'result': u'warning',
   u'started_at': u'2017-04-11 11:30:03 UTC',
   u'state': u'stopped',
   u'username': None},
  {u'cli_example': None,
   u'ended_at': u'2017-04-11 11:31:20 UTC',
   u'humanized': {u'action': u'Run hosts job:',
    u'errors': [u'A sub task failed'],
    u'input': u'Run ls',
    u'output': u'2 task(s), 0 success, 2 fail'},
   u'id': u'5d366568-e50a-4a01-bfca-766922b65950',
   u'input': {u'current_user_id': 3,
    u'job_category': u'Commands',
    u'job_invocation': {u'description': u'Run ls',
     u'id': 7,
     u'name': u'Commands'},
    u'job_invocation_id': 7,
    u'locale': u'en'},
   u'label': u'Actions::RemoteExecution::RunHostsJob',
   u'output': {u'failed_count': 2,
    u'pending_count': 0,
    u'success_count': 0,
    u'total_count': 2},
   u'pending': False,
   u'progress': 1.0,
   u'result': u'warning',
   u'started_at': u'2017-04-11 11:31:06 UTC',
   u'state': u'stopped',
   u'username': None}]}
```

svtkachenko commented on nailgun/entities.py:
```diff
@@ -4283,6 +4283,46 @@ def create(self, create_missing=None):
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'id': entity_fields.IntegerField(),
```
AFAIK, there is no need to specify `id` field directly.

svtkachenko commented:
@pondrejk Can you please also provide results for `read()` and not only `read_json()`?

pondrejk commented:
@svtkachenko thank you for the feedback, added path and unit test, read output:
```
In [4]: rec=entities.RecurringLogic(id=2)

In [5]: rec.read()
Out[5]: nailgun.entities.RecurringLogic(task_group_id=8, iteration=2, state=u'finished', end_time=None, cron_line=u'* * * * *', id=2)
```

abalakh commented on nailgun/entities.py:
```diff
@@ -4283,6 +4283,59 @@ def create(self, create_missing=None):
+            'end_time': entity_fields.DateTimeField(),
+            'iteration': entity_fields.IntegerField(),
+            'task_group_id': entity_fields.IntegerField(),
+            'state': entity_fields.StringField(),
```
Missing the `tasks` field which is present in read_json output. OneToManyField(ForemanTask), most likely.

pondrejk replied:
huh, sorry, I don't know where it got lost

abalakh commented on nailgun/entities.py:
```diff
@@ -4283,6 +4283,59 @@ def create(self, create_missing=None):
+            'cron_line': entity_fields.StringField(),
+            'end_time': entity_fields.DateTimeField(),
+            'iteration': entity_fields.IntegerField(),
+            'task_group_id': entity_fields.IntegerField(),
```
Isn't it an id of some entity?

pondrejk replied:
didn't  find any matching entity

svtkachenko commented:
ACK pending @abalakh comments

svtkachenko commented on nailgun/entities.py:
```diff
@@ -4283,6 +4283,59 @@ def create(self, create_missing=None):
 
 
+class RecurringLogic(
+        Entity,
```
Actually, one more comment: this now should fit one line.

pondrejk commented:
sorry for delay, I addressed the comments

```
rec=entities.RecurringLogic(id=2)
rec.read()
nailgun.entities.RecurringLogic(task_group_id=5, task=[nailgun.entities.ForemanTask(id=u'87c0c1f3-70fa-4658-be9f-da6e6a4ac596'), nailgun.entities.ForemanTask(id=u'0723a105-f6e4-45ba-ac25-63752bffc121')], iteration=2, state=u'finished', end_time=None, cron_line=u'*/6 * * * *', id=2)

```
